c/busybox-1.22.0/cal_false-unreach-call_true-no-overflow_true-valid-memsafety.c

```diff
@@ -1832,9 +1832,41 @@ ssize_t write(int fildes, const void *buf, size_t nbyte)
   return ret;
 }
 
+int getopt_long(int argc, char * const argv[], const char *optstring,
+                const struct option *longopts, int *longindex)
+{
+  (void)*longopts;
+  (void)longindex;
+
+  int result = -1;
+
```

Comment:
I believe that here should also be:
```C
if (optind == 0)
  optind = 1;
```

Code that use GNU extensions can set `optind` to 0 instead of 1 to reset the state before another pass through arguments. I know that this implementation does not check for any GNU extensions like '+', but without this piece of code the program can continue with `optind` being 0, which may lead to problems e.g. here:
https://github.com/sosy-lab/sv-benchmarks/blob/master/c/busybox-1.22.0/tee_true-no-overflow_true-valid-memsafety.c#L947, where the `names` variable can point before `argv` as the `optind` may be 0. That pointer is dereferenced on the line 1030, resulting in an invalid dereference.